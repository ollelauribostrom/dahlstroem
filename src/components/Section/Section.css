.section {
  padding: 20px 0;
}

.section__heading {
  margin-bottom: 5px;
  color: #6e297a;
  font-size: 16px;
}
