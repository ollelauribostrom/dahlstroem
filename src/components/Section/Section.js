import React from 'react';
import PropTypes from 'prop-types';

const Section = props => (
  <div className="section">
    <h3 className="section__heading">{props.title}</h3>
    {props.children}
  </div>
);

Section.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default Section;
