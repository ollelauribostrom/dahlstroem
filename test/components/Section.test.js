import React from 'react';
import { shallow } from 'enzyme';

import Section from '../../src/components/Section';

describe('Tests for Section', () => {
  it('renders correctly for a single child', () => {
    const wrapper = shallow(
      <Section title="test">
        <span>child</span>
      </Section>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly for a multiple children', () => {
    const wrapper = shallow(
      <Section title="test">
        <span>child</span>
        <span>child</span>
      </Section>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
