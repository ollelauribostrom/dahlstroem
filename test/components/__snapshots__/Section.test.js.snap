// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Section renders correctly for a multiple children 1`] = `
<div
  className="section"
>
  <h3
    className="section__heading"
  >
    test
  </h3>
  <span>
    child
  </span>
  <span>
    child
  </span>
</div>
`;

exports[`Tests for Section renders correctly for a single child 1`] = `
<div
  className="section"
>
  <h3
    className="section__heading"
  >
    test
  </h3>
  <span>
    child
  </span>
</div>
`;
